peekingduck/pipeline/nodes/draw/instance_mask.py

```diff
@@ -147,20 +117,81 @@ def run(self, inputs: Dict[str, Any]) -> Dict[str, Any]:
+            if self.config["effect"][effect] is not None:
+                if effect == "standard_mask" and self.config["effect"][effect]:
+                    output_img = self._draw_standard_masks(
+                        inputs["img"],
+                        inputs["masks"],
+                        inputs["bbox_labels"],
+                        inputs["bbox_scores"],
+                    )
+                    break  # only apply the first effect
+                if effect != "standard_mask":
+                    output_img = self._mask_apply_effect(
+                        inputs["img"],
+                        inputs["masks"],
+                        effect,
+                    )
+                    break  # only apply the first effect
+
         return {"img": output_img}
 
+    def _validate_configs(self) -> None:
+        self.check_valid_choice("instance_color_scheme", {"random", "hue_family"})
+
+        effects = (
+            "standard_mask",
+            "contrast",
+            "brightness",
+            "gamma_correction",
+            "blur",
+            "mosaic",
+        )
+        effects_count = 0
+        for effect, setting in self.config["effect"].items():
+            if effect not in effects:
+                raise ValueError(f"{effect} must be one of {effects}")
+            if setting:
+                effects_count += 1
+
+        if effects_count == 0:
+            raise ValueError("At least one effect must be enabled in the config.")
+        if effects_count > 1:
+            raise ValueError("Only one effect must be enabled at a time.")
+
+        self.check_valid_choice("effect_area", {"objects", "background"})
+
+        Node._check_type(self.config["effect"]["contrast"], float)
+        Node._check_number_range(self.config["effect"]["contrast"], "[0, 3]")
+        Node._check_type(self.config["effect"]["brightness"], int)
+        Node._check_number_range(self.config["effect"]["brightness"], "[-100, 100]")
+        Node._check_type(self.config["effect"]["gamma_correction"], float)
+        Node._check_number_range(self.config["effect"]["gamma_correction"], "[0, +inf]")
+        Node._check_type(self.config["effect"]["blur"], int)
+        Node._check_number_range(self.config["effect"]["blur"], "[1, +inf]")
+        Node._check_type(self.config["effect"]["mosaic"], int)
+        Node._check_number_range(self.config["effect"]["mosaic"], "[1, +inf]")
+        Node._check_type(self.config["contours"]["show"], bool)
+        Node._check_type(self.config["contours"]["thickness"], int)
+        Node._check_number_range(self.config["contours"]["thickness"], "[1, +inf]")
+
+    @staticmethod
+    def _check_type(var: Any, var_type: Any) -> None:
+        if var is not None and not isinstance(var, var_type):
+            raise ValueError(f"Config: '{var}' must be a {var_type.__name__} value.")
+
+    @staticmethod
+    def _check_number_range(var: Any, number_range: str) -> None:
```

Comment:
We cannot use the existing check_bounds() from ThresholdCheckerMixin because that was hard-coded to check variables at the "top level" of config e.g. config["source"], and cannot handle nested configs such as that required here e.g. config["effect"]["contrast"].

I have created another checker function here that can handle nested configs, regardless of how deep the nesting goes. The function can also handle new checks if we want to expand its capabilities later. For example, we can add a checker to do what ThresholdCheckerMixin's check_valid_choice() does.